File: application/core/src/main/java/br/com/beblue/bluerecords/core/usecase/GeneroUseCaseImpl.java
package br.com.beblue.bluerecords.core.usecase;

import br.com.beblue.bluerecords.core.entitidade.Genero;
import br.com.beblue.bluerecords.core.repositorio.GeneroRepositorio;

import java.util.List;

public class GeneroUseCaseImpl implements GeneroUseCase {

    public GeneroUseCaseImpl(GeneroRepositorio generoRepositorio) {
        this.generoRepositorio = generoRepositorio;
    }

    private GeneroRepositorio generoRepositorio;

    @Override
    public List<Genero> buscarGenero() {
        return generoRepositorio.buscarGenero();
    }
}

File: application/core/src/main/kotlin/br/com/beblue/bluerecords/core/usecase/genero/GeneroUseCaseImpl.kt
package br.com.beblue.bluerecords.core.usecase.genero

import br.com.beblue.bluerecords.core.entity.Genero
import br.com.beblue.bluerecords.core.repositorio.GeneroRepositorio

class GeneroUseCaseImpl(private val generoRepositorio: GeneroRepositorio) : GeneroUseCase
{
    override fun buscarGenero(): List<Genero> {
        return generoRepositorio.buscarGenero()
    }
}